### DISCUSSION TITLE
Helical/Spiral FDK Shadowing Artifacts

### DISCUSSION THREAD

**Discussion by pdsiCoop** [CERN/TIGRE#489] [Q&A] [answered]
Hello!

We’ve been trying to use TIGRE for the reconstruction of helical/spiral CT datasets using the FDK algorithm. We’re able to reconstruct helical data, but currently see fairly extreme shadowing artifacts around objects we'd like. As an example, I've attached below a slice from an ACR (American College of Radiology) phantom dataset we’ve reconstructed. It may also be relevant that these artifacts appear to shift or circle around while moving in scan direction. There also seems to be a gradient across the image that we think may have a similar underlying cause. 

![TIGRE_shaowing_example](https://github.com/CERN/TIGRE/assets/145499149/81399b2d-2edd-4a1e-bf19-772f5f5a9393)

Reconstructions using FDK with circular/axial data do not result in these artifacts. Further, if we use TIGRE’s ax function as a forward projector on the result of an axial reconstruction with a set of helical geometry parameters, and then reconstruct that data with TIGRE’s FDK, the streaks appear again. This might suggest these artifacts are related to how the FDK algorithm is implemented specifically while doing helical reconstructions. It seems like someone else may have encountered a similar issue here (https://github.com/CERN/TIGRE/issues/481), but it’s hard to be sure.

One attempt made at fixing these artifacts, with some success, is a row weighting scheme as suggested in this paper (https://doi.org/10.1088/0031-9155/49/11/007). This results in some improvement of the artifacts, although they still persist strongly. Below are the results of two row weightings with different parameters.

![TIGRE_weighted_0 6](https://github.com/CERN/TIGRE/assets/145499149/9a1628c6-a966-4408-be1f-78243a9d61c5)
![TIGRE_weighted_0 85](https://github.com/CERN/TIGRE/assets/145499149/bd2ea7e4-9600-4941-840a-ca0b63192f21)

Anyone have further thoughts on what could be causing these artifacts and how to fix them? 

Thanks

**Comment by AnderBiguri** [comment, marked as answer]
Hi @pdsiCoop ! If you are talking about helical CBCT, the unfortunate answer is that yes, FDK does not support this geometry.  This is an FDK issue, the way the algorithm is defined does not support a vertical detector shift. 

I am not aware if there is some modification to the algorithm weigths such that it corrects for helical data, I don't know. But generally speaking FDK only supports standard circular CBCT. Some modifications to support shifted detector exist (and are available in TIGRE), so I suspect there must be some publication somewhere about what to do for helical data. 

Unfortunately the current answer is that TIGRE FDK does not support it. Have you tried some iterative algorithm? That should work.

**Reply by pdsiCoop** [reply]
Hello,

Thanks for the fast reply! That is unfortunate to hear. We'll try looking around for anything about shifted detectors/modifications for helical.

In the meantime we'll also try reconstructing data with some of TIGRE's assorted iterative algorithms. We have tried CGLS which does not have these artifacts, at the cost of a significantly longer reconstruction time.

**Reply by AnderBiguri** [reply]
That's right, it's the thing with Iterative algorithm. They are almost always produce better results, but slower.

**Reply by EAM-Math** [reply]
just a side comment: I agree with @AnderBiguri.  I am working on a project where FDK (in circular trajectory of course) produces less (but different) artefacts than corresponding iterative reconstruction.  An exception to the rule.

**Reply by AnderBiguri** [reply]
@EAM-Math generally speaking indeed iteratives will do better, but realistically, given good enough data and simple enough geometry, FDK can do really good. E.g. in micro-CT applications, unless you are limited by scanning time, you can always acquire clean enough data that doing other than FDK is not worth the effort

**Comment by stephoog** [comment]
Hey! You could try using [LEAP](https://github.com/llnl/leap). It has helical FBP.

**Reply by AnderBiguri** [reply]
Great! you could also make a PR to change the FBP in TIGRE to accept helical, I assume is a small change ;)

**Comment by EAM-Math** [comment]
I'm just gettin started with TIGRE, but I do recall that FDK as originally formulated does not work for anything but circular acquisition trajectories.  However, I have a very-much related questions:   Does TIGRE allow for the use helical, or perhaps even double-helical geometries/trajectories?
